vm_supervisor/migrations/env.py

```diff
@@ -0,0 +1,70 @@
+from logging.config import fileConfig
+
+from alembic import context
+from vm_supervisor.conf import make_db_url
+from sqlalchemy import create_engine
+
+# # this is the Alembic Config object, which provides
+# # access to the values within the .ini file in use.
+# config = context.config
+#
+# # Interpret the config file for Python logging.
+# # This line sets up loggers basically.
+# if config.config_file_name is not None:
+#     fileConfig(config.config_file_name)
+
+# Auto-generate migrations
+from vm_supervisor.metrics import Base
+target_metadata = Base.metadata
+
+# other values from the config, defined by the needs of env.py,
+# can be acquired:
+# my_important_option = config.get_main_option("my_important_option")
+# ... etc.
+
+
+def run_migrations_offline() -> None:
+    """Run migrations in 'offline' mode.
+
+    This configures the context with just a URL
+    and not an Engine, though an Engine is acceptable
+    here as well.  By skipping the Engine creation
+    we don't even need a DBAPI to be available.
```

Comment:
This is very low level, how can a person new to the project understand the difference between offline and online ?

Reply:
RTFM :smile: This file is auto-generated by alembic. Basically online mode runs your Python migration scripts directly, whereas in offline mode you use alembic to generate SQL scripts and apply that however you like to your DB. We're only ever gonna use `online` mode in our case.